### ISSUE TITLE
Engintron breaking cpanel

### ISSUE THREAD

**Issue by rvencu** [engintron/engintron#760]
After one week of blissful speed, I started to get lots of 504 Gateway Timeouts. Especially when generating cache with WP Rocket. After this situation trying to disable Engintron took forever but it finalized. 

Modifying the timeouts in config to 500s did not solve anything, timeouts always appears under 60 seconds.

With it disabled everything works normally, also with normal Apache speed.

Trying to reinstall it from disabled mode via cPanel interface seems to never finish.

I am uninstalling it now from shell because the entire server is stuck on 504 gateway timeout.

**Comment by rvencu** [comment]
OK. It seems that after Engintron is uninstalled I still get the 540 errors on one domain only...

**Comment by fevangelou** [comment]
It would at least help others if you could describe what the problem was
and how you solved it.

**Comment by rvencu** [comment]
Yes, of course. I should think about it.

My setup involves WordPress and I am using WP Rocket as caching plugin and a push CDN. Whenever I delete the cache and rebuild it, I need to push static files to the CDN via rsync. My initial solution was to rsync the entire folder structure and exclude several file types such as php, etc. The rsync was triggered by inotify watch on source folders.

At some point in time I installed new software on the account that increased the number of static files. Therefore when I rebuilt the cache with a 250ms frequency the rsync processes were triggered too fast and execution took too long rasing processor usage up to 160.

It took some time to debug what was happening. After I improved the CDN rsync solution by installing lsyncd that is actually similar thing but it has the advantage to run as a service and also replicate only the changes. This is solving the CPU load and I can do again cahe rebuilding very fast.

Because Apache was not responsive the Nginx was triggering those errors.

I hope this helps someone, let me know if you like more details.
